raiden/tests/integration/api/test_pythonapi.py

```diff
@@ -19,6 +20,15 @@
 @pytest.mark.parametrize('number_of_nodes', [2])
 @pytest.mark.parametrize('number_of_tokens', [1])
 def test_token_addresses(raiden_network, token_addresses):
+    raise_on_failure(
```

Comment:
Hmm, that would introduce a new convention where the `test_*` functions would have this call and the `run_test_*` functions actually run the tests. I believe the better approach to go about this is to have `raise_on_failure` be a decorator function.

Reply:
I really don't want to go there and have to mess up with pytest's dependency injection. A decorator will return a function with a new signature, and the function signature is used to determine the tests dependencies.